python/ray/train/v2/api/config.py

```diff
@@ -89,6 +89,8 @@ class FailureConfig(FailureConfigV1):
     """
 
     fail_fast: Union[bool, str] = _DEPRECATED
+    # -1 means infinite reschedule.
+    resize_failure_limit: int = -1
```

Comment:
I think it's more intuitive for this configuration to be "maximum number of consecutive startup failures", rather than a global counter that keeps incrementing.

This is so that if resources are unavailable for a certain time period, you'll error out after 3 consecutive attempts for example. But failed startups separated by hours/days are probably not related and don't need to count toward the same failure limit.